### PR TITLE
Support Stringtype, Integertype as answer option type when item type is choice.

### LINKED ISSUE #1360: `answerOption.valueString` crashes SDC library
**Describe the bug**
Loading a questionnaire with `answerOption.valueString` instead of `answerOption.valueCoding` as the library [expects](https://github.com/google/android-fhir/blob/master/datacapture/src/main/java/com/google/android/fhir/datacapture/MoreAnswerOptions.kt#L28-L29) causes a crash. valueStrings and other types are perfectly legitimate options [per the spec](https://www.hl7.org/fhir/questionnaire-definitions.html#Questionnaire.item.answerOption.value_x_), however.

**Component**
SDC library & reference app.

**To Reproduce**
1. Add Questionnaire below to the app
2. View Questionnaire

**Additional context**
Sample questionnaire:

```json
{
  "resourceType": "Questionnaire",
  "id": "3141",
  "url": "http://example.org/fhir/Questionnaire/1",
  "title": "Questionnaire with answerOption.valueString",
  "status": "draft",
  "item": [
    {
      "linkId": "FIRSTBC",
      "type": "choice",
      "text": "Indicate if this is first breast cancer or new cancer on contralateral or ipsilateral breast:",
      "answerOption": [
        {
          "valueString": "Primary tumor"
        },
        {
          "valueString": "New ipsilateral"
        },
        {
          "valueString": "New contralateral"
        }
      ]
    }
  ]
}
```

### PR DESCRIPTION
**Pull request by santosh-pingle**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

Fixes #1360 

**Description**

- Support `IntegerType` and `StringType` as answer option type when item type is `Choice`.
- QuestionnaireResponse will always have answerOption type as Coding when item type is choice.

**Alternative(s) considered**
Have you considered any alternatives? And if so, why have you chosen the approach in this PR?

**Type**
Choose one: Bug fix

**Screenshots (if applicable)**

**Checklist**
- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/docs/CODE_OF_CONDUCT.md).
- [x] I have read the [Contributing](https://github.com/google/android-fhir/wiki/Contributing) page.
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate ).
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach.
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project.
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally.
- [x] I have built and run the demo app(s) to verify my change fixes the issue and/or does not break the demo app(s).

### CODE REVIEW

**Review by jingtang10 — CHANGES_REQUESTED** [review]
can you please add `integer|date|time|string|Coding|Reference(Any)` as per the spec.

**Review by jingtang10 — CHANGES_REQUESTED** [review]
Santosh can you also implement this: https://github.com/google/android-fhir/issues/1360#issuecomment-1157550094

**Comment by jingtang10** [comment]
@santosh-pingle this is now a high priorty item - thanks for fixing this.

**Comment by santosh-pingle** [comment]
in progress.

File: datacapture/src/main/java/com/google/android/fhir/datacapture/MoreAnswerOptions.kt [hunk 1]
```diff
@@ -18,21 +18,34 @@ package com.google.android.fhir.datacapture
...
+      is StringType -> (value as StringType).getLocalizedText() ?: value.toString()
+      is Coding -> {
+        val display = valueCoding.displayElement.getLocalizedText() ?: valueCoding.display
+        if (display.isNullOrEmpty()) {
+          valueCoding.code
+        } else {
+          display
+        }
+      }
+      else -> throw IllegalArgumentException("$value is not supported.")
```
Comment (jingtang10) [inline datacapture/src/main/java/com/google/android/fhir/datacapture/MoreAnswerOptions.kt:48]: test this

**Review by jingtang10 — CHANGES_REQUESTED** [review]

File: datacapture/src/main/java/com/google/android/fhir/datacapture/MoreAnswerOptions.kt (hunk 1 above)
Comment (santosh-pingle) [reply datacapture/src/main/java/com/google/android/fhir/datacapture/MoreAnswerOptions.kt:48]: To test else part, it is required to set value of `QuestionnaireItemAnswerOptionComponent` with not supported type, if you do so then fhir library api throws error.
` throw new Error("Not the right type for Questionnaire.item.answerOption.value[x]: "+value.fhirType());`

Comment (jingtang10) [reply datacapture/src/main/java/com/google/android/fhir/datacapture/MoreAnswerOptions.kt:48]: ok so the `QuestionnaireItemAnswerOptionComponent` class doesn't allow you to set a type that's not one of the types here. so it's sort of non-strict type safety. got it.

**Review by jingtang10 — APPROVED** [review]
thanks @santosh-pingle

**Pull request merged**